ui/events/event_auto_repeat_handler.cc

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ui/events/event_auto_repeat_handler.h"
+
+#include "base/threading/thread_task_runner_handle.h"
+#include "ui/events/base_event_utils.h"
+
+namespace ui {
+
+namespace {
+
+const int kRepeatDelayMs = 500;
+const int kRepeatIntervalMs = 50;
+
+}  // namespace
+
+EventAutoRepeatHandler::EventAutoRepeatHandler() : weak_ptr_factory_(this) {
+  repeat_delay_ = base::TimeDelta::FromMilliseconds(kRepeatDelayMs);
+  repeat_interval_ = base::TimeDelta::FromMilliseconds(kRepeatIntervalMs);
+}
+
+EventAutoRepeatHandler::~EventAutoRepeatHandler() {}
+
+void EventAutoRepeatHandler::SetCallback(
+    DispatchAutoRepeatEventCallback callback) {
```

Comment:
pass by reference?

Reply:
and can we pass the callback to the ctor instead? Just as it was before.

Reply:
> pass by reference?

I think this is how other callbacks are also passed too.

> and can we pass the callback to the ctor instead? Just as it was before.

I tried, but got a crash (passing Bind(&Class::Method, base::Unretained(this)) from initialization just :/